src/ocean/io/select/protocol/task/TaskSelectClient.d

```
@@ -161,9 +160,9 @@ class TaskSelectClient: ISelectClient
         try
         {
             this.task = task.getThis();
-            assert(this.task);
+            verify(this.task !is null);
```

In cases like this we have to be careful: `verify(object !is null)` does not call the object invariant while `assert(object)` does. `Task` doesn't have an invariant so it's fine here, but in general this change can silently drop a debugging aid.

---

So the better thing to do would be:
```D
// old
assert(object);

// replace with
verify(object !is null && object); // call invariant
```

---

No, only literally `assert(object)` works. It is a magic expression to call the invariant.

---

And `assert(task)` is perfectly fine here:
1) it will just crash on the next line in release build
2) it protects against being called from main thread context - not something that may not be noticed before production deloyment